hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -667,6 +681,12 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
             if (now > timeToStartJoin) {
                 startJoin();
             }
+            if (node.isMaster()) {
+                long startTime = clusterClock.getClusterStartTime();
+                nodeEngine.getOperationService().invokeOnTarget(SERVICE_NAME,
```

Comment:
Yes, but ```FinalizeJoinOperation``` is sent by every node to newly joining member, what I want is only master node sending its clusterId and startTime to the newly joining node. I tried to implement using only ```FinalizeJoinOperation```  and removed the extra operation from the PR. During join process master node, adds an extra ```FinalizeJoinOperation```  like this in ```startJoin``` method.  

```
for (MemberInfo member : setJoins) {
    if (node.isMaster()) {
        //Only master node should send clusterId and startTime
        long startTime = clusterClock.getClusterStartTime();
        calls.add(invokeClusterOperation(new FinalizeJoinOperation(memberInfos, postJoinOp, time,
                clusterId, startTime), member.getAddress()));
    }
    calls.add(invokeClusterOperation(new FinalizeJoinOperation(memberInfos, postJoinOp, time), member.getAddress()));
}
```  
Also inside run method of  ```FinalizeJoinOperation```  newly joined sets clusterId and startTime like this,

```
if (isClusterInfoUpdateOperation()) {
    // Master node sends extra finalize join operation
    // to set clusterId and startTime.
    clusterService.setClusterId(clusterId);
    clusterService.getClusterClock().setClusterStartTime(clusterStartTime);
    return;
}
```